src/backend/common/CMakeLists.txt

```diff
@@ -0,0 +1,51 @@
+
+
+add_library(afcommon_interface INTERFACE)
+
+target_sources(afcommon_interface
+  INTERFACE
+    ${CMAKE_CURRENT_SOURCE_DIR}/ArrayInfo.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/ArrayInfo.hpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/FFTPlanCache.hpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/InteropManager.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/InteropManager.hpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/MatrixAlgebraHandle.hpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/MemoryManager.hpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/MersenneTwister.hpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/SparseArray.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/SparseArray.hpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/cblas.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/constants.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/defines.hpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/dim4.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/dispatch.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/dispatch.hpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/err_common.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/err_common.hpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/host_memory.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/host_memory.hpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/sparse_helpers.hpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/util.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/util.hpp
+    ${PROJECT_BINARY_DIR}/version.hpp
+  )
+
+target_include_directories(afcommon_interface
+  INTERFACE
+    ${CMAKE_CURRENT_SOURCE_DIR}
+    ${PROJECT_BINARY_DIR}
+  )
+
+add_library(afcommon_lapack_interface INTERFACE)
```

Comment:
If we guard it then the library will not be created and we will have to guard everywhere the library is used. This way the library is created either way but in case `LAPACK_FOUND` is not set then it doesn't do anything so we can add this library anywhere without special guards.